### ISSUE TITLE
Experiment Module: Add location field to experiment upload form

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#259]
In the experiment module we have a data field which is for the (arable) field name. Initially I was slightly confused as to why this wasn't auto-filling with the (arable) field names, but today realised it is because the data field it isn't actually a location. I think we initially specified as a simple text field because the location attributes of experiments were quite complex to visualise, and it was the easiest thing to do quickly. However, now we have a few examples, I think there is an obvious structure. for the lcoation heirarchy which I have been doing manually. I'll do my best to explain...

Currently, when creating an experiment, the plots are given a location by the GeoJSON upload. The experiment upload then automatically creates a land asset for the experiment surrounds (to be renamed experiment boundary, see #256). The experiment surrounds has the option to add a geometry (usually added later) and there is also the option to add a parent asset. This is initially blank when the experiment is created, but I have been editing them so that the parent asset is the field (or fields for were some experiments are split over multiple fields). 

In short the location structure for an experiment should be:

Arable Field (grandparent - set as parent in experiment surrounds)
Experiment Surrounds (aka boundary) (asset in experiment plan?)
Plots (child) (parent is the experiment surrounds)

So what is listed as 'Field' in the experiment module is actually the field location, and it would be quite nice if this replaced the current 'Field' data field. See for example the below for Broadbalk

It's definitely quite complicated. We might want to simplify it a bit. :)

![image](https://user-images.githubusercontent.com/16349781/184666193-2d394545-bafd-42c6-9f55-9a398a4dab09.png)

**Comment by paul121** [comment]
@aislinnpearson would it be helpful if the "Upload experiment" page let the user select which "grandparent" arable field the experiment is a part of? It could automatically assign the Experiment Boundary as a child of that field.

What you're describing makes sense, it sounds like we just need to formalize this a bit more? This additional question could be required or optional on the upload page.

**Comment by aislinnpearson** [comment]
> @aislinnpearson would it be helpful if the "Upload experiment" page let the user select which "grandparent" arable field the experiment is a part of? It could automatically assign the Experiment Boundary as a child of that field.

Yes! That would be perfect! The only thing I'd ask (and this is where it gets complicated in the locations) is if we could add _multple_ fields, for experiments which are split over more than one. A rare occurance, but it does happen sometimes for the larger experiments.

**Comment by paul121** [comment]
Adding multiple fields to the parent, no prob :+1:

**Comment by paul121** [comment]
![Screenshot from 2022-09-19 13-19-03](https://user-images.githubusercontent.com/3116995/191108923-e0c633aa-c198-44dc-81f6-fac5166e4f5f.png)

Done! You can now specify the experiment boundary parents via this upload form. Right now this is an optional field but we could make it required. @aislinnpearson repopen this if you would like to make it required or adjust the wording in the description.

**Comment by aislinnpearson** [comment]
Hi @paul121,

I'm happy to leave it as optional for the time being. From next year the workflow will be different - plot layouts will be decided before the field location, so we will want more of a workflow approach... and as I am doing all the uploads this year I'll make sure I enter it, and then we can decide at what point in the process it needs to be required. :)
